src/middlewared/middlewared/plugins/disk.py

```diff
@@ -791,6 +791,12 @@ def label_to_dev(self, label, geom_scan=True):
         disks_pairs = [disks for disks in list(serials.values())]
         disks_pairs.sort(key=lambda x: int(x[0][2:]))
 
+        # If its TrueNAS, no multipath already exists but new multipath were detected
+        # we should not continue. Its likely there is wrong cabling in the system.
+        # See #42042 for details.
+        if not is_freenas and not mp_disks and any(map(lambda x: len(x) > 1, disks_pairs)):
+            return 'BAD_CABLING'
```

Comment:
Why not to do logging here? Unexpected failures in configuration may occur not only on booting but also maybe on hotplug

Reply:
In this case it can only happen on boot.

Logging to console and /var/log/messages within middlewared is not in place nor trivial for a 11.1-U7 change.

Reply:
I think we can do it this way in oldstable and stable release and do it "properly" (if we decide so) in devel branch.

Another reason for what I am offering as "proper" is having as less logic as possible in bash scripts.

Reply:
We can do it "properly" once we move it away from bash script and use etc plugin. Lets leave it like that for now, please.